tls/s2n_config.c

```diff
@@ -439,6 +439,7 @@ int s2n_config_set_status_request_type(struct s2n_config *config, s2n_status_req
 
     POSIX_ENSURE_REF(config);
     config->status_request_type = type;
+    config->status_request_type_set = type == S2N_STATUS_REQUEST_OCSP;
 
```

Comment:
Shouldn't this always be true? We just care that the API was called at all.

Reply:
That's what I had originally, but I realized there was a case like this:
https://github.com/aws/s2n-tls/blob/dbe22c6eec3a6586fc759e0f144e280ded929e36/tests/unit/s2n_config_test.c#L644-L646

which would allow `s2n_config_set_verification_ca_location` to toggle the status. But I guess this is kind of a weird case. Would it be better to just always set this to true?

Reply:
We don't want s2n_config_set_verification_ca_location to ever toggle the status for the server though, right? Aren't we trying to remove the strange legacy behavior?

Reply:
Correct, `s2n_config_set_verification_ca_location` should never toggle the status for servers. The problem with setting `status_request_type_set` to true when setting REQUEST_NONE is that `s2n_config_set_verification_ca_location` can then toggle the status to REQUEST_OCSP. When the config is set on the connection,  `status_request_type_set` will be true, and the status will be REQUEST_OCSP, so there's no way to know if it was actually enabled intentionally or not.

I added a comment to the test case. Let me know if it's still unclear, or if I'm missing something!